impl/src/main/java/com/okta/sdk/impl/api/OAuth2TokenClientCredentialsResolver.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2020-Present Okta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.okta.sdk.impl.api;
+
+import com.okta.sdk.authc.credentials.ClientCredentials;
+
+import java.util.Optional;
+
+/**
+ * @since 1.6.0
+ */
+public class OAuth2TokenClientCredentialsResolver implements ClientCredentialsResolver {
```

Comment:
This class shouldn't be needed, when you set the AuthScheme == `NONE`

Reply:
We still need this class to set a credentials resolver for token client. _BaseClient_ requires credentials resolver to be set for all implementations (see https://github.com/okta/okta-sdk-java/blob/68333b61c9fa096c557a022347bd4e0f2ea05521/impl/src/main/java/com/okta/sdk/impl/client/BaseClient.java#L45 & https://github.com/okta/okta-sdk-java/blob/68333b61c9fa096c557a022347bd4e0f2ea05521/impl/src/main/java/com/okta/sdk/impl/ds/DefaultDataStore.java#L106). So, let's retain it.

Reply:
If that is the case, rename the class to indicate its usage (NoOpClientCredentialsResolver or something), or just use the `DefaultClientCredentialsResolver` and pass `() -> Optional.empty()` into the constructor?